core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -238,10 +237,7 @@ public MavenInstallation getMaven() {
         return null;
     }
 
-    private Object readResolve() throws ObjectStreamException {
-        if (injectBuildVariables == null) {
-            injectBuildVariables = true;
-        }
```

Comment:
Oleg's comment about `readResolve` also applies here. Do not break `readResolve()` functionality. See [Hint on retaining backward compatibility](https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility). On a side note, that content really needs to be modernized and moved to the [Jenkins Developer Reference](https://www.jenkins.io/doc/developer/book/) on jenkins.io. Volunteers would be appreciated, but this is not related to your change. The only thing necessary for this PR is to preserve existing `readResolve()` functionality.

Reply:
Thanks for pointing that out, I'll take a look at the docs. If I find time I will create a PR for that.